libtransact/src/scheduler/serial/execution.rs

```diff
@@ -54,11 +54,9 @@ impl Iterator for SerialExecutionTaskIterator {
                     }
                 }
             }
-            Err(err) => {
-                error!(
-                    "failed to send request for next in execution task iterator: {}",
-                    err
-                );
+            Err(_) => {
+                // This is expected when the scheduler's internal thread has shutdown because it has
+                // no more execution tasks.
```

Comment:
Instead of ignoring errors (here and above), we should switch to properly sending sentinels which tell us when the other side has shutdown. This way, we still log when this is an error (like the other side crashed or did something unexpected that didn't cleanly shutdown).

Reply:
I removed this commit and the one that created a new sender for each request; those issues will be addressed in a separate PR.